File: src/main/java/com/group/InternMap/Model/User/Student.java
package com.group.InternMap.Model.User;

import jakarta.persistence.Entity;

import java.io.Serializable;
import java.util.Calendar;

@Entity
public class Student extends User implements Serializable {

    private int graduatingYear = Calendar.getInstance().get(Calendar.YEAR) + 1;
    private String uniName;
    private String studentMajor;
    private String faculty;

    public Student(String fName, String lName, String email, String plainPassword, int graduatingYear, String uniName, String studentMajor, String faculty) {
        super(fName, lName, email, plainPassword);
        this.graduatingYear = graduatingYear;
        this.uniName = uniName;
        this.studentMajor = studentMajor;
        this.faculty = faculty;
    }

    public Student() {}

    public int getGraduatingYear() {
        return graduatingYear;
    }

    public void setGraduatingYear(int graduatingYear) {
        this.graduatingYear = graduatingYear;
    }

    public String getUniName() {
        return uniName;
    }

    public void setUniName(String uniName) {
        this.uniName = uniName;
    }

    public String getStudentMajor() {
        return studentMajor;
    }

    public void setStudentMajor(String studentMajor) {
        this.studentMajor = studentMajor;
    }

    public String getFaculty() {
        return faculty;
    }

    public void setFaculty(String faculty) {
        this.faculty = faculty;
    }
}

File: Frontend/app/Model/Student.ts
class Student extends User {

    graduatingYear: number;
    studentMajor: String;
    faculty: String;
    uniName: String;
    cv: CV;

    constructor(graduatingYear: number, studentMajor: String, faculty: String, uniName: String, cv: CV, email: String, fname: String, lname: String, role: String, id: bigint) {
        super(email, fname, lname, role, id);
        this.graduatingYear = graduatingYear;
        this.studentMajor = studentMajor;
        this.faculty = faculty;
        this.uniName = uniName;
        this.cv = cv;
    }
}